src/extension/intellisenseHelper.ts

```
@@ -53,21 +69,53 @@ export class IntellisenseHelper {
+        if (IntellisenseHelper.isSalsaSupported()) {
+            // this is the correct version of vscode, which includes TypeScript (Salsa) support, nothing to do here
+            return Q.resolve<boolean>(isRestartRequired);
+        }
+
+        return IntellisenseHelper.getInstalledTypeScriptVersion()
+            .then(function(installProps: InstallProps) {
+
+                if (installProps.installed === false) {
+
+                    if(semver.neq(IntellisenseHelper.s_typeScriptVersion, installProps.version)) {
+                        Log.logInternalMessage(LogLevel.Debug, "TypeScript is installed with the wrong version: " + installProps.version);
+                    }
+                    else {
+                        Log.logInternalMessage(LogLevel.Debug, "TypeScript is not installed");
+                    }
+                    return Q.resolve<boolean>(true);
                 }
 
-                return isRestartRequired;
+                Log.logInternalMessage(LogLevel.Debug, "Installed TypeScript version is correct");
+                return Q.resolve<boolean>(false);
+            })
+            .then( (install: boolean) => {
+
+                if(install) {
+                    let installPath: string = path.resolve(IntellisenseHelper.getUserHomePath(), ".vscode");
+                    let installParam: string = "npm install --prefix " + installPath + " typescript@" + IntellisenseHelper.s_typeScriptVersion;
+                    Log.logInternalMessage(LogLevel.Debug, "Installing TypeScript: " + installParam);
+                    return Q.nfcall(child_process.exec, installParam)
+                        .then(() => {
+                            return Q.resolve<boolean>(true);
+                        })
+                        .catch( (err: any) => {
+                            Log.logError("Error attempting to install TypeScript: " + err);
+                            return Q.resolve<boolean>(isRestartRequired);
```

Should we be attempting to recover from this failure, or should we let it fail the "prepare the project for salsa" sequence and report that failure at a higher level?